app/src/main/java/org/breezyweather/sources/CommonConverter.kt

```diff
@@ -673,48 +678,190 @@ private fun completeHalfDayFromHourlyList(
 
     val newHalfDay = initialHalfDay ?: HalfDay()
 
+    val extremeTemperature = if (newHalfDay.temperature?.temperature == null
+        || newHalfDay.temperature.apparentTemperature == null
+        || newHalfDay.temperature.windChillTemperature == null
+        || newHalfDay.temperature.wetBulbTemperature == null) {
+        getHalfDayTemperatureFromHourlyList(newHalfDay.temperature, halfDayHourlyList, isDay)
+    } else newHalfDay.temperature
+
+    val totalPrecipitation = if (newHalfDay.precipitation?.total == null) {
+        getHalfDayPrecipitationFromHourlyList(halfDayHourlyList)
+    } else newHalfDay.precipitation
+
+    val maxPrecipitationProbability = if (newHalfDay.precipitationProbability?.total == null) {
+        getHalfDayPrecipitationProbabilityFromHourlyList(halfDayHourlyList)
+    } else newHalfDay.precipitationProbability
+
+    val maxWind = if (newHalfDay.wind?.speed == null) {
+        getHalfDayWindFromHourlyList(halfDayHourlyList)
+    } else newHalfDay.wind
+
+    val avgCloudCover = newHalfDay.cloudCover ?: getHalfDayCloudCoverFromHourlyList(halfDayHourlyList)
+
+    val avgVisibility = halfDayHourlyList.mapNotNull { it.visibility }.let {
+        if (it.isNotEmpty()) it.average() else null
+    }
+
     var halfDayWeatherText = newHalfDay.weatherText
     var halfDayWeatherPhase = newHalfDay.weatherPhase
     var halfDayWeatherCode = newHalfDay.weatherCode
 
     // Weather code + Weather text
     if (halfDayWeatherCode == null || halfDayWeatherText == null) {
-        // Update at 12:00 on daytime and 00:00 on nighttime
-        val halfDayHourlyListWeather = halfDayHourlyList
-            .firstOrNull { it.date.time == dailyDate.time + (if (isDay) 12 else 24) * 3600 * 1000 }
-            ?: halfDayHourlyList.first() // Use first element in the list to avoid null
+        // best representative hour we strive to find
+        var hour: HourlyWrapper? = null
+        // custom weather code to override the day code if needed
+        var customWeatherCode: WeatherCode? = null
+        // ideal time (12:00 for day and 24:00 for night)
+        val bestTime = Date(dailyDate.time + (if (isDay) 12 else 24).hours.inWholeMilliseconds)
+        // sort by time elapsed from the ideal temporal hour
+        val sorted = halfDayHourlyList.sortedBy { abs(bestTime.time - it.date.time) }
```

Comment:
What is the purpose? All hours from half day are equally important

Reply:
Removed